### ISSUE TITLE
Train error

### ISSUE THREAD

**Issue by Kriz5178** [liuruijin17/LSTR#21]
I'd like to ask when I run train.py by thread=4 and batch size change to 4  
Ram memory still  cost to 16G then computer crashed
Can I ask ur PC format?

**Comment by liuruijin17** [comment]
Hi, please also set the "chunk_sizes": [4] if you use batch_size 4 and train on a single GPU.

On my PC, Ubuntu 16.04, 1080Ti, 64G RAM, when thread=4, batch_size=4, and chunk_sizes=4, ~2G RAM memory will be used.
I run  `free -h` before training, and the terminal shows the used Mem is 10G. After I run the training code a while, I run `free -h` again, and the terminal shows the used Mem is 12G. Meanwhile, the GPU Memory Usage is 765MiB.

**Comment by Kriz5178** [comment]
Hi, the last problem was solved by adding ram to 24G.
I am trying to use curve lane dataset for training, mainly modified the data reading method is as follows.
I modified it according to the reading method of culane.
    def _extract_data(self):

        max_lanes = 0
        image_id  = 0

        self._old_annotations = {}
       # print(self.anno_files)
        for anno_file in self.anno_files:
            
            with open(anno_file, 'r') as anno_obj:
                annolines = anno_obj.readlines()
            for annoline in annolines:
                rel_path = annoline[:-1]
                img_path = self.root +'/'+ rel_path
                anno_path = img_path.replace('jpg', 'lines.json').replace('images', 'labels')
                try:
                    strlanes = json.load(open(anno_path))
                    lanes       = []
                    y_sampleses = []
                    gt_lanes    = []
                    # for strlane in strlanes:
                    for strlane in strlanes['Lines']:
                        x_lane_point = []
                        y_lane_point = []
                        for i in strlane:
                            x_value=float(i['x'])
                            y_value=int(float(i['y']))
                            if 0 < x_value < 2560 and 0 < y_value < 1440:
                                x_lane_point.append(x_value)
                                y_lane_point.append(y_value)
                        y_sampleses.append(y_lane_point)
                        gt_lanes.append(x_lane_point)
                        lanes.append([(x, y) for (x, y) in zip(x_lane_point, y_lane_point) if x >=0])
                    if not gt_lanes:  # this image has no fucking lanes
                        continue
                    lanes = [lane for lane in lanes if len(lane) > 0]
                    max_lanes = max(max_lanes, len(lanes))
                    self.max_lanes = max_lanes
                    self.max_points = max(self.max_points, max([len(l) for l in gt_lanes]))
                    self._image_file.append(img_path)
                    self._image_ids.append(image_id)
                    self._old_annotations[image_id] = {
                        'path': img_path,
                        'org_path': rel_path,
                        'org_lanes': gt_lanes,
                        'lanes': lanes,
                        'aug': False,
                        'y_samples': y_sampleses
                    }
                    image_id += 1
                    if image_id % 1000 == 0:
                        print('Processing the {} th image...'.format(image_id))
                except FileNotFoundError:
                    pass
                    continue
But when I run train.py, I meet the problem that tensor is empty in tensor.
I had tested the data reading method should be no problem.
Is there any tips when I change the dataset?

thx.

**Comment by liuruijin17** [comment]
Sorry for being late, tough months about lots of deadlines.

The script seems fine. 
Which tensor is empty? image or the labeling points? Maybe you can visualize the data flow to locate the problems.
